main.py

```
@@ -5,8 +5,36 @@
+        groq_api_key: Provided API key. If None, reads from environment.
+
+    Returns:
+        Validated API key string.
+
+    Raises:
+        MissingAPIKeyError: If no valid API key is found.
+    """
+    if groq_api_key is None:
+        groq_api_key = os.getenv("GROQ_API_KEY")
+    if not groq_api_key:
+        raise MissingAPIKeyError("GROQ_API_KEY is required but not provided or set in environment.")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)